### ISSUE TITLE
Add features to analyze barren plateaus

### ISSUE THREAD

**Issue by ChenZhao44** [QuantumBFS/ZXCalculus.jl#76]
[Analyzing the barren plateau phenomenon in training quantum neural networks with the ZX-calculus](https://arxiv.org/abs/2102.01828)

Computing variance of gradients in variational quantum circuits will help one to design circuit structure.
It is possible to implement these features with the current data structure. But the better way is to use the ZX-calculus with ground notation. This is more elegant to handle mixed states.

**Comment by exAClior** [comment]
This is implemented in #93, see tests. I will close this for now.
